tests/migration/recovery.go

```diff
@@ -0,0 +1,238 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package migration
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/net/context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	k8score "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/libkubevirt/config"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libnet/job"
+	"kubevirt.io/kubevirt/tests/libpod"
+	"kubevirt.io/kubevirt/tests/libregistry"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libvmops"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Migration recovery", decorators.SigCompute, func() {
+	DescribeTable("should successfully defer a migration", func(fakeSuccess, flakeCheck bool) {
+		virtClient, err := kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		if flakeCheck {
+			// Skip the test meant for the flake-checker if we're not currently flake-checking
+			if !flags.FlakeCheck {
+				Skip("This version of the test is for the flake-checker, skipping.")
+			}
+			kv := getCurrentKvConfig(virtClient)
+			kv.VMStateStorageClass = "local"
+			config.UpdateKubeVirtConfigValueAndWait(kv)
```

Comment:
So, this is something I'm experimenting with. I'm trying to avoid serial tests as much as possible because they are expensive.
When running on sig-compute, the CR is not updated and we rely on the fact that sig-compute doesn't do RWX FS.
However, the default storage class on the flake-checker lane does, so we need to force RWO by modifying the CR, hence to two versions of the test.